ttljob: add controls to pause TTL jobs

otan opened the pull request:
See individual commits for details.

otan committed:
ttljob: add cluster setting for disabling TTL job

This commit disables TTL job execution if the CLUSTER SETTING
sql.ttl.job.enabled is set to false.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/76835)
<!-- Reviewable:end -->

otan committed:
ttljob: introduce ttl_pause

This commit introduces ttl_pause as a table storage parameter, pausing
TTL job for only the given table.

Release note: None

rafiss commented on pkg/sql/ttl/ttljob/ttljob.go:
```diff
@@ -68,8 +70,14 @@ var (
 		0,
 		settings.NonNegativeInt,
-	).WithPublic()
+	)
```
was public removed on purpose?

otan replied:
ah bad rebase conflict resolution!

rafiss commented on pkg/sql/ttl/ttljob/ttljob.go:
nit: could use `jobEnabled.Key()` so we don't need the extra const

rafiss commented on pkg/sql/ttl/ttljob/ttljob.go:
where do background job errors like this show up? somewhere in the console?

otan replied:
`SHOW JOBS` displays these

rafiss reviewed (approved):
lgtm! had some optional nits and a question

otan commented:
thanks!

bors r=rafiss

otan commented:
bors r-

otan committed:
fixup! ttljob: add cluster setting for disabling TTL job

otan commented:
bors r=rafiss
